paginate-nfts/pages/index.js

```diff
@@ -0,0 +1,123 @@
+import Head from 'next/head';
+import styles from '../styles/Home.module.css';
+import { Metaplex } from '@metaplex-foundation/js';
+import { clusterApiUrl, Connection, PublicKey } from '@solana/web3.js';
+import { useEffect, useState } from 'react';
+
+const connection = new Connection(clusterApiUrl('devnet'));
+const mx = Metaplex.make(connection);
+
+export default function Home() {
+  const [address, setAddress] = useState(
+    'EAqjUWVX2m9fdfGNBzTY5zSiid1Sb9V3x6EL8ssZBTkw',
+  );
+
+  const [nftList, setNftList] = useState(null);
+  const [loading, setLoading] = useState(false);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [currentView, setCurrentView] = useState(null);
+  const perPage = 1;
+
+  useEffect(() => {
+    if (nftList) {
+      const execute = async () => {
+        const startIndex = (currentPage - 1) * perPage;
+        const endIndex = currentPage * perPage;
+        await loadData(startIndex, endIndex);
+        setCurrentView(nftList.slice(startIndex, endIndex));
+        setLoading(false);
+      };
+      execute();
+    }
+  }, [nftList, currentPage]);
+
+  const loadData = async (startIndex, endIndex) => {
+    const nftsToLoad = nftList.filter((nft, index) => {
+      return (
+        index >= startIndex && index < endIndex && nft.metadataTask.isPending()
+      );
+    });
+
+    const promises = nftsToLoad.map((nft) => nft.metadataTask.run());
+    await Promise.all(promises);
+  };
+
+  const fetchNFTs = async () => {
```

Comment:
Could we move the `fetchNFTs` function above the useEffect hook? That way it reads in the same order as the tutorial.